java/core/src/main/java/sleeper/core/deploy/LambdaJar.java

```diff
@@ -33,63 +34,74 @@ public class LambdaJar {
+            .artifactId("compaction-task-creation").add();
+    public static final LambdaJar PARTITION_SPLITTER = builder().filenameFormat("lambda-splitter-%s.jar")
+            .imageName("partition-splitter-lambda")
+            .artifactId("splitter-lambda").add();
+
     // The query module includes Hadoop, which makes the jar too big to deploy directly.
     // It seems difficult to reduce this significantly, but this may be unnecessary if we switch to using DataFusion
     // for queries.
-    public static final LambdaJar QUERY = withFormatAndImageDeployWithDocker("query-%s.jar", "query-lambda");
-    public static final LambdaJar CUSTOM_RESOURCES = withFormatAndImage("cdk-custom-resources-%s.jar", "custom-resources-lambda");
-    public static final LambdaJar METRICS = withFormatAndImage("metrics-%s.jar", "metrics-lambda");
-    public static final LambdaJar STATESTORE = withFormatAndImage("statestore-lambda-%s.jar", "statestore-lambda");
+    public static final LambdaJar QUERY = builder().filenameFormat("query-%s.jar")
+            .imageName("query-lambda")
+            .artifactId("query-lambda")
+            .alwaysDockerDeploy(true).add();
+    public static final LambdaJar CUSTOM_RESOURCES = builder().filenameFormat("cdk-custom-resources-%s.jar")
+            .imageName("custom-resources-lambda")
+            .artifactId("cdk-custom-resources").add();
+    public static final LambdaJar METRICS = builder().filenameFormat("metrics-%s.jar")
+            .imageName("metrics-lambda")
+            .artifactId("metrics").add();
+    public static final LambdaJar STATESTORE = builder().filenameFormat("statestore-lambda-%s.jar")
+            .imageName("statestore-lambda")
+            .artifactId("statestore-lambda").add();
 
     private final String filename;
     private final String imageName;
     private final boolean alwaysDockerDeploy;
     private final String filenameFormat;
-
-    private LambdaJar(String filenameFormat, String version, String imageName, boolean alwaysDockerDeploy) {
-        this.filenameFormat = Objects.requireNonNull(filenameFormat, "filename must not be null");
-        this.filename = String.format(filenameFormat, version);
-        this.imageName = Objects.requireNonNull(imageName, "imageName must not be null");
-        this.alwaysDockerDeploy = Objects.requireNonNull(alwaysDockerDeploy, "alwaysDockerDeploy must not be null");
+    private final String artifactId;
+
+    private LambdaJar(Builder builder) {
+        this.filenameFormat = requireNonNull(builder.filenameFormat, "filename must not be null");
+        this.filename = String.format(filenameFormat, SleeperVersion.getVersion());
+        this.imageName = requireNonNull(builder.imageName, "imageName must not be null");
+        this.alwaysDockerDeploy = requireNonNull(builder.alwaysDockerDeploy, "alwaysDockerDeploy must not be null");
+        this.artifactId = requireNonNull(builder.artifactId, "Artifact Id must not be null");
```

Comment:
The capitalisation here is inconsistent with the other fields. If we use English style, I think it should be "ID" rather than "Id".